libraries/hotstuff/qc_chain.cpp

```diff
@@ -182,35 +183,28 @@ namespace eosio::hotstuff {
       return b;
    }
 
-   bool qc_chain::evaluate_quorum(const extended_schedule& es, const hs_bitset& finalizers, const fc::crypto::blslib::bls_signature& agg_sig, const hs_proposal_message& proposal) {
-
+   bool qc_chain::evaluate_quorum(const hs_bitset& finalizers, const fc::crypto::blslib::bls_signature& agg_sig, const hs_proposal_message& proposal) {
 
       if (positive_bits_count(finalizers) < _pacemaker->get_quorum_threshold()){
          return false;
       }
 
       fc::crypto::blslib::bls_public_key agg_key;
 
+      std::vector<fc::crypto::blslib::bls_public_key> pks =_pacemaker->get_finalizer_keys();
+
```

Comment:
Should `EOS_ASSERT` that `pks.size()` == `finalizers.size()`

Reply:
As per issue 1626 (https://github.com/AntelopeIO/leap/issues/1626), this requires us to resize the hs_bitset to the number of expected finalizers for the generation corresponding to the qc we're evaluating. I perform this resizing in the quorum_certificate constructor that uses a quorum_certificate_message as parameter, as this seems to be the best time to do it. Since other quorum_certificate initialization-related functions are still called with a hardcoded 21 parameter, this is also what I did here (with the associated todo). Once we are set on a safe finalizer set transition algorithm and appropriate data structures, we can use the correct active finalizer size as dictated by context.